JS/Lesson14/onder/script.js

```diff
@@ -9,71 +9,99 @@ const randomNumberFrom0To10 = Math.floor(Math.random()*10+1);
 const stringExercise1 = 'Object-oriented programming is a powerful paradigm.';
+console.log("Exercise1: ", stringExercise1.length);
+
+console.log('Object-oriented programming is'.length);
 
 console.log('------------------------');
 // 2. Use the `slice` method to extract 'Debugging code' from the string below and print it to the console.
 const stringExercise2 = 'Debugging code can be a frustrating but necessary task.';
+console.log("exercise2: ", stringExercise2.slice(0,14));
 
 console.log('------------------------');
 // 3. Use the `toUpperCase` method to convert a string to uppercase and print it to the console.
 const stringExercise3 = 'Python is a popular language for data analysis and machine learning.';
-
+console.log("exercise3: ",stringExercise3.toUpperCase());
 console.log('------------------------');
 // 4. Use the `toLowerCase` method to convert a string to lowercase and print it to the console.
 const stringExercise4 = 'FUNCTIONAL PROGRAMMING EMPHASIZES PURE FUNCTIONS AND IMMUTABLE DATA.';
+console.log("Exercise4: ",stringExercise4.toLowerCase());
 
 console.log('------------------------');
 // 5.a Use the `indexOf` method to find the index of "l" character in the string stringExercise5 and print it to the console.
 const stringExercise5 = 'JavaScript is a versatile language used for web development and beyond.';
+console.log("Exercise5: ",stringExercise5.indexOf('l'));
 
 console.log('------------------------');
 // 5.b Use the `indexOf` method to find the index of "language" in the string stringExercise5 and print it to the console.
-
+console.log("Exercise5.b: ", stringExercise5.indexOf('language'));
 console.log('------------------------');
 // 6. Use the `lastIndexOf` method to find the last index of of "l" character in the string stringExercise5 and print it to the console.
-
+console.log("Exercise6: ", stringExercise5.lastIndexOf('l'));
 console.log('------------------------');
 // 7. Use the `charAt` method to retrieve first, 10th, 3rd from the end and the last characters from the string stringExercise7 and print them to the console.
 const stringExercise7 = 'Git is a popular version control system used by developers.';
+console.log("Exercise7: ", stringExercise7.charAt(0));
+console.log("Exercise7: ", stringExercise7.charAt(9));
+console.log("Exercise7: ", stringExercise7.charAt(stringExercise7.length-3));
 
 console.log('------------------------');
 // 8. Use the `split` method to split a string into an array of substrings and print it to the console.
 const stringExercise8 = 'Algorithms are a fundamental part of computer science.';
-
+console.log("Exercise8: ", stringExercise8.split(" "));
 console.log('------------------------');
 // 9. Use the `join` method to join an array of strings into a single string and print it to the console.
 const arrayExercise9 = ["An", "array",  "of", "strings"];
+console.log("Exercise9: ", arrayExercise9.join("-"));
```

Comment:
If you want to make a sentence, it's better to use a space (`join(" ")` as a connector :)